src/devices/depthCamera/depthCameraDriver.cpp

```diff
@@ -139,8 +144,8 @@ depthCameraDriver::~depthCameraDriver()
         delete m_depthFrame;
     if (m_imageFrame)
         delete m_imageFrame;
-    if (m_cameraDescription)
-        delete m_cameraDescription;
+    if (m_paramParser)
```

Comment:
You might want to refactor the destructor like this:
```c++
delete m_depthFrame;
delete m_imageFrame;
delete m_paramParser;
```
This is doable since `nullptr` pointers won't be deleted.